src/Integrations/Integrations/LaravelQueue/LaravelQueueIntegration.php

```diff
@@ -46,6 +45,25 @@ public function init()
     {
         $integration = $this;
 
+        \DDTrace\hook_method(
+            'Illuminate\Queue\Worker',
+            'kill',
+            function () {
+                // span.c:ddtrace_close_all_open_spans() -> ensure the span is closed rather than dropped
+                ini_set('datadog.autofinish_spans', '1');
+
+                dd_trace_close_all_spans_and_flush();
+                dd_trace_internal_fn('synchronous_flush', 3);
```

Comment:
Maybe this should be promoted to a proper API function then, if there are use cases outside of testing :-)